core/templates/dev/head/player/AnswerClassificationService.js

```diff
@@ -0,0 +1,142 @@
+// Copyright 2015 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Classification service for answer groups.
+ *
+ * @author wxyxinyu@gmail.com (Xinyu Wu)
+ */
+
+oppia.factory('answerClassificationService', [
+    '$http', '$q', 'learnerParamsService', function(
+      $http, $q, learnerParamsService) {
+
+  /**
+   * Finds the first answer group with a rule that returns true. This should be
+   * synced with classify() in reader.py.
+   *
+   * @param {*} answer The answer that the user has submitted.
+   * @param {array} answerGroups The answer groups of the interaction. Each
+   *     answer group contains rule_specs, which is a list of rules.
+   * @param {object} defaultOutcome The default outcome of the interaction.
+   * @param {function} interactionRulesService The service which contains the
+   *     rules of that interaction.
+   *
+   * @return {object} An object representing the answer group with the
+   *     following properties:
+   * <ul>
```

Comment:
Ooh, interesting -- what syntax is this? (I like it, but just curious about the lack of closing li tags and wondering about the second opening ul tag.